src/components/ArcMap.vue

```diff
@@ -0,0 +1,102 @@
+<template>
+  <div id='map'></div>
+</template>
+
+<script>
+// Future Implementation:
+// * Users can register events until they reach a max/min of total 'street length'
+//   this standardizes that each volunteer does similar amounts of work.
+
+import { loadModules } from 'esri-loader';
+// import { expression } from '@babel/template';
+
+export default {
+  /* eslint-disable no-alert, no-unused-vars */
+  name: 'ArcMap',
+  mounted() {
+    // lazy load the required ArcGIS API for JavaScript modules and CSS
+    const reserveSegment = {
+      title: 'Reserve',
+      id: 'reserve-this',
+      image: 'https://developers.arcgis.com/javascript/latest/sample-code/popup-actions/live/Measure_Distance16.png',
+    };
+    const template = {
+      // autocasts as new PopupTemplate()
+      title: '{ST_NAME} {ST_TYPE}', // Show attribute value
+      content: '<b>ID:</b> {SEGMENT_ID} <br><b>Length:</b> {SHAPElen} ft<br><b>Available?</b> Yes<br><b>Last Observered:</b> August 17, 2018',
+      actions: [reserveSegment],
+    };
+    loadModules(['esri/Map', 'esri/views/MapView', 'esri/layers/FeatureLayer', 'esri/layers/GraphicsLayer',
+      'esri/Graphic'],
+    { css: true }).then(([ArcGISMap, MapView, FeatureLayer, GraphicsLayer]) => {
+      const map = new ArcGISMap({
+        basemap: 'topo-vector',
+      });
+      this.view = new MapView({
+        container: this.$el,
+        map,
+        center: [-71.0892, 42.3398],
```

Comment:
Are these Boston coordinates?

Reply:
Can you just write a little more documentation in this file so its easier to know whats going on?